pupil_src/shared_modules/annotations.py

```diff
@@ -188,6 +190,38 @@ def fire_annotation(self,annotation_label):
         self.annotations_list.append(notification)
         self.annotations_by_frame[notification['index']].append(notification)
 
+    @classmethod
+    def csv_representation_keys(self):
+        return ('label', 'timestamp','duration','source','index')
+
+    @classmethod
+    def csv_representation_for_annotations(self, annotation):
+        return (
+            annotation['label'],
+            annotation['timestamp'],
+            annotation['duration'],
+            annotation['source'],
+            annotation['index']
+        )
+
+
+    def export_annotations(self,export_range,export_dir):
+
+        if not self.annotations:
+            logger.warning('No annotations in this recording nothing to export')
+            return
+
+        annotations_in_section = chain(*self.annotations_by_frame[export_range])
+        annotations_in_section = dict([(a['index'],a) for a in annotations_in_section]).values() #remove dublicates
```

Comment:
One could use a dict-comprehension in here. That could be copied to the old code as well.